api/client.go

```diff
@@ -139,12 +139,22 @@ func (c *Client) FromPing(resp *Ping) *Client {
 	return NewClient(c.logger, conf)
 }
 
+type Header struct {
+	Name  string
+	Value string
+}
+
 // NewRequest creates an API request. A relative URL can be provided in urlStr,
 // in which case it is resolved relative to the BaseURL of the Client.
 // Relative URLs should always be specified without a preceding slash. If
 // specified, the value pointed to by body is JSON encoded and included as the
 // request body.
-func (c *Client) newRequest(ctx context.Context, method, urlStr string, body interface{}) (*http.Request, error) {
+func (c *Client) newRequest(
+	ctx context.Context,
+	method, urlStr string,
+	body interface{},
+	headers ...Header,
+) (*http.Request, error) {
```

Comment:
this method signature is probably getting to the point where it's worth factoring it into a request struct or something:
```Go
type Request struct {
	Method  string
	URL     string
	Body    any
	Headers []Header
}

// ...

func (c *Client) newRequest(ctx context.Context, req Request) (*http.Request, error) { 
	// ...
}
```
... but for now i don't think we need to change it, and it would mean a bunch refactoring in everything that uses this method (which is basically everything in this package)

Reply:
Yeah, we specifically chose this array of headers signature so that we didn't have to change every method call. I'll make a ticket to do this refactor.